src/main/java/org/testng/xml/internal/XmlSuiteUtils.java

```diff
@@ -7,47 +7,36 @@
 import org.testng.xml.XmlSuite;
 import org.testng.xml.XmlTest;
 
+import java.util.Iterator;
 import java.util.List;
 import java.util.Set;
 
 /**
- * A utility class that exposes helper methods to work with {@link XmlSuite}
+ * A utility class to work with {@link XmlSuite}
  */
 public final class XmlSuiteUtils {
```

Comment:
I don't catch the changes of this class. Could you add some unit tests that explain the behavior?

Reply:
I thought I added `testCloneIfContainsTestsWithNamesMatchingAnyChildSuites` and `testIfTestnamesComesFromDifferentSuite` from `org.testng.xml.internal.XmlSuiteUtilsTest` to cover all the new features I added. 
Could you tell me which part I missed or make your confuse?